src/libstore/s3_binary_cache_store.go

```diff
@@ -0,0 +1,131 @@
+package libstore
+
+import (
+	"context"
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/minio/minio-go/v7"
+	"io"
+	"log"
+	"net/url"
+	"os"
+)
+
+type S3Store struct {
+	Profile    string
+	Region     string
+	Scheme     string
+	bucketName string
+	client     *minio.Client
+}
+
+// SDK should be in environment
+func (s *S3Store) getUri() string {
+	return "s3://" + s.bucketName
+}
+
+func (s *S3Store) FileExists(ctx context.Context, path string) (bool, error) {
+	found, err := s.client.BucketExists(ctx.Background(), s.bucketName)
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+	if found {
+		obj, err := s.client.GetObject(ctx, s.bucketName, path, minio.GetObjectOptions{})
+		if err != nil {
+			log.Fatalln(err)
```

Comment:
In general, client libraries should avoid containing logging as it is hard to remove it. It's best to return the error object instead and let the parent deal with it.